### PR TITLE
ENH: laser beam transport motion system (btms)

### LINKED ISSUE #1049: Add support for the Laser Beam Transport Motion system
## Context
As detailed in the concept of operations documents linked below,
* [Laser Beam Transport Protection System](https://docs.google.com/document/d/1VnhIy3L4WVfWmfMqut5xda-WDZKL5WO4Zig3-Cbz-nc/view) (BTPS)
* [Laser Beam Transport Motion System](https://docs.google.com/document/d/1_VPrnwoOBjTgWnHpTgTX-byykokMmCoRpb1Yqoek-FM/view) (BTMS)

A simulation IOC is available in [sim-ioc](https://github.com/pcdshub/sim-ioc/pull/12) that allows for this entire system to be tested offline.

## To be added

`pcdsdevices` should provide an ophyd `Device` interface to all aspects of the btps/btms for hutch-python control.
Its device representations will also be used for:
* PV information for the higher-level/more user-friendly [GUI](https://github.com/pcdshub/btms-ui)
* A high-level verification of motion layer / sanity-checker
    * A control-system agnostic logic checker should be added for the purposes of decoupling some of the logic from the `Device` class and also for the test suite
* Test suite for verification

## To be changed
Changes will be required in order to support this system:
* An initial BTPS implementation exists in pcdsdevices but requires some updates:
    * Synchronization with the PV naming adjustments made in https://github.com/pcdshub/lcls-plc-las-bts (https://github.com/pcdshub/lcls-plc-las-bts/pull/13)
    * Screens need updating for this as well
* Addition of new BTMS signals where appropriate in existing BTPS classes

### PR DESCRIPTION
**Pull request by klauer**
## Description
* Includes BTPS (laser beam transport protection system) PV updates 
* Preliminary code for checking for motion conflicts (more on this later)

## Motivation and Context

Closes #1049

* PLC code update will be rolled out the week of September 9th
* Simulation IOC PR: https://github.com/pcdshub/sim-ioc/pull/12
* PLC code PR: https://github.com/pcdshub/lcls-plc-las-bts/pull/13
* BTMS user interface: https://github.com/pcdshub/btms-ui

## How Has This Been Tested?
Included test suite

## Where Has This Been Documented?
* Beam Transport System Motion Automation Concept of Operations
* Beam Transport Protection System Concept of Operations

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+exclude = .git,__pycache__,build,dist,versioneer.py,pcdsdevices/_version.py,docs/source/conf.py
```
Comment (ZLLentz) [inline .flake8:2]: Compare with the previous exclude list:
https://github.com/pcdshub/pcdsdevices/blob/ce56c58a9fcb7e047a2efbdcbd179838e9970884/setup.cfg#L8

Comment (klauer) [reply .flake8]: Thaank you

**Review by slactjohnson — APPROVED** [review]
Reviewed together with Ken, LGTM

File: pcdsdevices/lasers/btps.py [hunk 2]
```diff
@@ -1,8 +1,36 @@
...
+from .btms_config import (BtmsSourceState, BtmsState, DestinationPosition,
+                          MoveError, SourcePosition)
+
+
+class BtpsVGC(VGC):
+    """
+    VGC subclass with 'valve_position' component added.
+    """
+    # TODO: this may be pushed into ValveBase, but need to check with others
+    # first
```
Comment (ZLLentz) [inline pcdsdevices/lasers/btps.py:26]: REF: https://github.com/pcdshub/pcdsdevices/pull/1044
We could absolutely merge 1044 I think

**Review by ZLLentz — APPROVED** [review]
I have less context than Tyler but it looks good to me too

**Comment by klauer** [comment]
Merging while I have approval and passing CI for this.
Warning ahead of time: there will almost certainly be a follow-up PR...

**Pull request merged**